profiles/views.py

```diff
@@ -61,28 +59,35 @@ class ProfileList(ListCreateAPIView):
      include_all: bool.
     """
     serializer_class = ProfileSerializer
-    permission_classes = (IsAuthenticatedOrReadOnly, )
+    permission_classes = [IsAuthenticatedOrReadOnly & SavedCompaniesListPermission]
     pagination_class = ForumPagination
 
     def get_queryset(self):
+        query_params_serializer = QueryParamSerializer(data=self.request.query_params)
```

Comment:
I told you about FilterQueryParamsSerializer, so you should have something like this here:

filters = FilterQueryParamsSerializer(data=self.request.query_params.get("filters"))